### PR TITLE
Shopify create smart collection

### LINKED ISSUE #1879: [ACTION] Shopify - Admin API - CreateSmartCollection
Shopify - Admin API - CreateSmartCollection

https://shopify.dev/api/admin-rest/2021-10/resources/smartcollection#[post]/admin/api/2021-10/smart_collections.json

### PR DESCRIPTION
**Pull request by andrewjschuang**
Shopify New Action

- [X] Create Smart Collection

Resolves #1879

### CODE REVIEW

**Comment by dylburger** [comment]
Thanks for submitting this PR! When we review PRs, we follow the [Pipedream component guidelines](https://pipedream.com/docs/components/guidelines/). If you're not familiar, here's a quick checklist:

- [ ] [Create components to address specific use cases whenever possible](https://pipedream.com/docs/components/guidelines/#component-scope)
- [ ] [Component `key`s should follow the format `app_name_slug-slugified-component-name`](https://pipedream.com/docs/components/guidelines/#component-key-pattern)
- [ ] Components should follow the [standard directory structure](https://pipedream.com/docs/components/guidelines/#folder-structure)
- [ ] [Prefer Node.js client libraries to REST APIs](https://pipedream.com/docs/components/guidelines/#using-apis-vs-client-libraries)
- [ ] When making API requests, [handle pagination](https://pipedream.com/docs/components/guidelines/#pagination) to ensure all data / events are processed
- [ ] [Use `secret` props](https://pipedream.com/docs/components/guidelines/#capturing-sensitive-data) to capture sensitive data
- [ ] Props and methods should be defined in [app files](https://pipedream.com/docs/components/guidelines/#app-files) whenever possible
- [ ] Document methods with [JS Docs](https://pipedream.com/docs/components/guidelines/#methods)
- [ ] [Use `optional` props](https://pipedream.com/docs/components/guidelines/#optional-vs-required-props) whenever possible, and [set a `default` value](https://pipedream.com/docs/components/guidelines/#default-values) where you can
- [ ] Use [async options](https://pipedream.com/docs/components/guidelines/#async-options) to accept user input wherever possible
- [ ] [Create subscription webhooks sources (vs polling sources) whenever possible](https://pipedream.com/docs/components/guidelines/#webhook-vs-polling-sources)
- [ ] [Always emit a `summary` with events](https://pipedream.com/docs/components/guidelines/#emit-a-summary)
- [ ] [Use a built-in deduping strategy wherever possible](https://pipedream.com/docs/components/guidelines/#deduping)
- [ ] When building webhook sources, [follow these best practices](https://pipedream.com/docs/components/guidelines/#webhook-sources)
- [ ] When building polling sources, [follow these best practices](https://pipedream.com/docs/components/guidelines/#polling-sources)

**Review by vellames-turing — DISMISSED** [review]
Looks great, nothing to comment on. Approved!

**Comment by andrewjschuang** [comment]
Hey @vellames-turing, can you take a look at this again? I've changed the way the rules prop selection works in this action

File: components/shopify/actions/common/rules.mjs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    "starts_with",
+    "ends_with",
+    "contains",
+    "not_contains",
+  ],
+};
+
+export {
+  rules,
+};
```
Comment (jcortes) [inline components/shopify/actions/common/rules.mjs:27]: Optional:
As far as I know we alwas try to use `export default` as an standard
 ```suggestion
export default rules;
```

**Review by jcortes — APPROVED** [review]
Hi @andrewjschuang I added jus a minor optional change. Everything else looks good to me. Ready for QA

**Comment by DilanAthukorala** [comment]
Verification completed successfully

**Comment by andrewjschuang** [comment]
@vellames-turing @jcortes can someone approve this please? I just noticed right now that this still isn't merged

**Comment by vellames-turing** [comment]
> @vellames-turing @jcortes can someone approve this please? I just noticed right now that this still isn't merged

I'd say to wait @jcortes approval since he did the first review

File: components/shopify/actions/create-smart-collection/create-smart-collection.mjs [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    rules: {
+      type: "integer",
+      label: "Number of rules",
+      description: "The number of rules to input",
+      default: 1,
+      min: 1,
+      reloadProps: true,
+    },
+  },
+  async additionalProps() {
```
Comment (vellames-turing) [inline components/shopify/actions/create-smart-collection/create-smart-collection.mjs:43]: -OPTIONAL-
I think we can use `reduce` and return it directly here. Sounds more elegant to me. But completely optional :)

**Review by vellames-turing — APPROVED** [review]
Hi there, thanks for your PR. Approved!

**Pull request merged**